File: doc/src_OpenGL/RubbishGL.cpp
// RubbishGL.cpp: implementation of the RubbishGL class.
//
//////////////////////////////////////////////////////////////////////
#include "Stdafx.h"
#include "OpenGL.h"

#include "Inkludy.h"

#ifdef _DEBUG
#undef THIS_FILE
static char THIS_FILE[]=__FILE__;
#define new DEBUG_NEW
#endif

//////////////////////////////////////////////////////////////////////
// Construction/Destruction
//////////////////////////////////////////////////////////////////////

RubbishGL::RubbishGL()
{
	list = 20;
}

RubbishGL::~RubbishGL()
{

}

void RubbishGL::draw(WorldObject* object)
{
	float mx;
	rubbish = (Rubbish*)object;
	glPushMatrix();
	mx = Direction::computeRotation(rubbish->getDirection());
	glRotatef(mx,0.0f,0.0f,1.0f); 
	glCallList(list);
	glPopMatrix();
}

void RubbishGL::init()
{
	glNewList(list, GL_COMPILE);
#include "rubbish"
	glEndList();
}

File: app/src/main/java/robot/RubbishGL.java
package robot;

import robot.object.WorldObject;
import robot.object.opengl.ObjectGL;

import javax.media.opengl.GL;

public class RubbishGL extends ObjectGL {

    Rubbish rubbish;
    public RubbishGL() {
        list = 20;
    }

    public void draw(GL gl, WorldObject object)
    {
        float mx;
        rubbish = (Rubbish)object;
        gl.glPushMatrix();
        mx = Direction.computeRotation(rubbish.getDirection());
        gl.glRotatef(mx,0.0f,0.0f,1.0f);
        gl.glCallList(list);
        gl.glPopMatrix();
    }

    public void init(GL gl)
    {
        gl.glNewList(list, GL.GL_COMPILE);
//#include "rubbish"
        gl.glEndList();
    }

}